### ISSUE TITLE
Blank DPS Lines

### ISSUE THREAD

**Issue by mayonegg** [lunyx/CasualMeter#89]
Frequently CasualMeter bugs out and just displays a blank line.

Can be seen from this player's meter: http://puu.sh/mipa9/d3786d29ce.jpg

**Comment by lunyx** [comment]
Does this actually happen when you're actively playing?  I've noticed it happen to me when I'm tabbed out doing something else and coming back, so I suspect it may just not be updating the UI constantly while it's inactive.

**Comment by mayonegg** [comment]
Yeah, it happens during both occasions of me idling and dungeoning. I do switch tabs frequently so it might be related. There are some times, however, where I reset the meter and it immediately produces a blank line while recording.

**Comment by kevorkien** [comment]
this has been happening to me constantly. In my case, the top player is always shown and all other players below are planked.

I play full screen and have Cas.Meter off on my 2nd monitor.
I have options set to Use Compact View, and Show Personal DPS.

While all the names underneath the top entry are there and can be clicked on to show a players dps breakdown (including name), all lines are blank. I will try to identify further conditions that produce this. Whether it happens after alt tabbing etc.

**Comment by lunyx** [comment]
I've had multiple people report this, but I've yet to be able to reproduce it recently.  I know I've definitely seen it before, but it just happened with such low frequency for me that it was impossible to figure out.  If you can figure out how to reliably reproduce it, that would be extremely helpful.

**Comment by kevorkien** [comment]
strangely enough for the first time (that I recall anyway) I had only 1 line missing, instead of the remainder. I'll try do those tests fo ry ou.

**Comment by kevorkien** [comment]
Ok I think i've isolated... or at least narrowed down the issue. It's compact view

**Bug Description:  _DPS entries beneath the topmost DPS appearing blank_**
**Bug type:**  Repeatable
**Initial Conditions:**
- Party of 3 players
- Size and Opacity: Max both
- Ignore One-Shots Checked (did not try without)
- Set Compact View to OFF

**Steps to Reproduce:**
   1. Configure Casual-Meter as Initial Conditions.
   2. DPS stuff
   3. Finish encounter
   4. Save the encounter
   5. Repeat so as to have 2 encounters saved
   6. DPS data should appear fine
   7. Open options, and check to Use Compact View
   8. DPS data should still appear fine
   9. Swap between saved encounters
10. DPS data will show all lines missing below the topmost

**Notes:**
- Tested against the following options ON/OFF with no variation to the above results:
 - _Auto-Save Encounter, Count Only Party, Count Only Bosses, Show Personal DPS_
- Only tested in one party of 3
- Was unable to produce the result of a single or multiple lines missing, between other lines
    (as mentioned in my previous post, I can only recall having this issue once though)

**Test PC setup.**
   -Windows 10-64
   -Dual Monitor Display, Casual Meter on 2nd Monitor.
   -Only tested in Full Screen Windowed mode as I was multi-boxing the 3 chars

**Please let me know if you want any other log files etc**

**Regards,
Aeon**

![casual meter bug sample pic](https://cloud.githubusercontent.com/assets/11830655/17103382/63c71852-52c1-11e6-91ac-ddb2e8c57256.jpg)
_Bug Mockup (I forgot to screenie one)_

**Comment by lunyx** [comment]
Thanks, I'll give this a shot when I can grab some people with time on their hands.

**Comment by lunyx** [comment]
I think I finally figured this one out.  I pushed out 1.2.3.0 that should address the problem.  

Developer note:  The core of the issue seems to be due to hotswapping the DPS listing templates, which may have invoked this Microsoft bug: https://support.microsoft.com/en-us/kb/2461678.  Unfortunately, the hotfix dll is no longer available to test, so I can't say for certain if it was caused by this, but all the symptoms matched, along with what another Microsoft representative mentioned on the msdn forums.  In any case, I worked around it by instead having a single view that has both types, but will just swap visibility status when the conditions are met.
